txscript/data/script_tests.json

```diff
@@ -1365,87 +1459,193 @@
+["1 1 BOOLAND", "NOP", "NONE", "OK", "BOOLAND must perform a logical and operation on the boolean values of the first two stack items"],
 ["1 0 BOOLAND", "NOT", "NONE", "OK"],
 ["0 1 BOOLAND", "NOT", "NONE", "OK"],
 ["0 0 BOOLAND", "NOT", "NONE", "OK"],
-["16 17 BOOLAND", "NOP", "NONE", "OK"],
-["1 1 BOOLOR", "NOP", "NONE", "OK"],
+["1 2147483648 BOOLAND", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
+
+["BOOLOR test coverage"],
+["NOP BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two input values and NOP must not be treated as one"],
+["BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two input values, stack has no items"],
+["0 BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two items on the stack, stack has no items"],
+["1 1 BOOLOR", "NOP", "NONE", "OK", "BOOLOR must perform a logical and operation the boolean values of the first two stack items"],
 ["1 0 BOOLOR", "NOP", "NONE", "OK"],
 ["0 1 BOOLOR", "NOP", "NONE", "OK"],
 ["0 0 BOOLOR", "NOT", "NONE", "OK"],
-["16 17 BOOLOR", "NOP", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUAL", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK"],
-["11 10 1 ADD", "NUMNOTEQUAL NOT", "NONE", "OK"],
+["1 2147483648 BOOLOR", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
+
+["NUMEQUAL test coverage"],
+["NOP", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack and NOP must not be treated as one"],
+["", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack, stack has no items"],
+["-1", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack, stack has one item"],
+["11 10 1 ADD", "NUMEQUAL", "NONE", "OK", "NUMEQUAL must compare the first numeric stack item to the second numeric stack item and check if they are equal"],
+["-200 200", "NUMEQUAL NOT", "NONE", "OK"],
+["-1 -1", "NUMEQUAL", "NONE", "OK"],
+["1", "0x02 0x0100 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Numerically equal but rejected by minimal data encoding requirements"],
+["11", "0x4c 0x03 0x0b0000 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0", "0x01 0x80 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Zero numerically equals negative zero but rejected by minimal data encoding requirements"],
+["0", "0x02 0x0080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Zero numerically equals negative zero with padding but rejected by minimal data encoding requirements"],
+["0x03 0x000080", "0x04 0x00000080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0x03 0x100080", "0x04 0x10000080 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["0x03 0x100000", "0x04 0x10000000 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA"],
+["1 2147483648 NUMEQUAL", "NOT", "NONE", "ERR_OUT_OF_RANGE", "NUMEQUAL must fail with stack item >4 bytes"],
```

Comment:
Same here regarding needing another test for the other operand being >4 bytes and comment.